rewrite-java/build.gradle.kts

```diff
@@ -64,6 +64,7 @@ dependencies {
     testRuntimeOnly(project(":rewrite-java-17"))
     testImplementation("com.tngtech.archunit:archunit:1.0.1")
     testImplementation("com.tngtech.archunit:archunit-junit5:1.0.1")
+    testImplementation("org.junit-pioneer:junit-pioneer:2.0.0")
```

Comment:
I would rather use JUnit 5's built in assertThrows() unless there is some really compelling advantage to `@ExpectedToFail`

Reply:
I've seen it in other modules, and I like the semantics of it for documenting a not-yet-implemented feature with a test.
But yeah, it does require another dep, and we can get similar functionality with assertThrows. I can swap it out when I'm at my desk later.

Reply:
I also quite like `@ExpectedToFail` for tests which test something that is supposed to work but doesn't due to a bug. I use `assertThrows()` when I want to test that something is supposed to throw an exception.

When the bug is fixed, all I have to do is to remove the `@ExpectedToFail` annotation.

Reply:
sounds like @sambsnyd is outvoted here, unless I hear otherwise :)
any other review concerns for this PR, before we rebase-not-squash and merge?
@timtebeek , @knutwannheden